### PR TITLE
Create .wasm file for metadata exchange plugin.

### PR DESCRIPTION
**Pull request by jplevyak**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**: This is a WIP. I would like to get some feedback on how to improve the process.

**Release note**:

```release-note
```

### CODE REVIEW

File: src/envoy/http/metadata_exchange/plugin.cc [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+#include "plugin.h"
+#include <google/protobuf/io/zero_copy_stream_impl_lite.h>
+#include "absl/strings/str_cat.h"
+#include "absl/strings/str_split.h"
+
+// clang-format off
+constexpr char CHAR_TABLE[] =
+    "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/";
+
+constexpr unsigned char REVERSE_LOOKUP_TABLE[256] = {
```
Comment (mandarjog) [inline src/envoy/http/metadata_exchange/plugin.cc]: I think this can go into a separate header / cc file.
It clutters the main file too much .

Comment (kyessenov) [reply src/envoy/http/metadata_exchange/plugin.cc]: Agree.

**Comment by brian-avery** [comment]
Can this be merged into master instead? We're cutting the 1.3 final soon and this is described as a WIP.

**Comment by brian-avery** [comment]
This should be merged into master first.

**Pull request closed without merging**